events-api/src/modules/events/events.controller.ts

```diff
@@ -23,3 +25,85 @@ EventsController.get(
     });
   }
 );
+
+EventsController.post(
+  "/",
+  validateRequestBody(eventsBodyDataDtoSchema),
+  async (req, res) => {
+    const requestBody = req.body as unknown as EventsBodyDataDto;
+
+    const event = await EventsService.newEvent(requestBody);
+
+    return res.status(201).json({
+      message: "Event created successfully",
+      data: event,
+      requestBody,
+    });
+  }
+);
+
+EventsController.patch(
+  "/:eventId",
+  validateSearchParams(eventsUpdateDtoSchema),
+  async (req, res) => {
+    const getEventID = eventIDSchema.parse(req.params['eventId']) as string;
+    const requestBody = req.body as unknown as EventsUpdateDto;
+
+    const updatedEvent = await EventsService.updateEvent(getEventID, requestBody);
+
+    return res.status(200).json({
+      message: "Event updated successfully",
+      data: updatedEvent
+    });
+  }
+);
+
+EventsController.delete(
+  "/:eventId",
+  async (req, res) => {
+    try {
+      const eventID = eventIDSchema.parse(req.params['eventId']) as string;
+      const deletedEvent = await EventsService.deleteEvent(eventID);
+      console.log(deletedEvent);
+      return res.status(204).json({
+        message: "Event deleted successfully",
+        data: deletedEvent
+      });
+
+    } catch (error) {
+      if (error instanceof z.ZodError) {
```

Comment:
Such error handling should be done on the level of the either validation middleware or global error handler